### PR TITLE
Update copy on In-App Marketplace

### PR DESCRIPTION
**Pull request by Dan-Q**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Updates copy in the new In-App Marketplace to fit the latest design. Specific changes are listed below. Changes overlapping with/treading on the toes of [WCCOM#17713](https://github.com/Automattic/woocommerce.com/issues/17713) are marked with a *:

- Updated footer text and swapped padlock icon for shield icon
- Updated color of "Woo Marketplace" footer icon/text *
- Fixed heading font size *
- Corrected placeholder text in search box *
- On "no results" page, improved semantics by using a `<h3>` rather than just bold style to indicate the heading
- On "no results" page, corrected color of body text

### How to test the changes in this Pull Request:

1. Build WooCommerce Admin on your local environment e.g.: cd plugins/woocommerce-admin then nvm use && pnpm start.
2. Visit [WooCommerce > Marketplace > Extensions](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions&tab=extensions) and view each of the items listed above

### CODE REVIEW

File: plugins/woocommerce-admin/client/marketplace/components/footer/footer.tsx [hunk 1]
```diff
@@ -23,18 +23,18 @@ const refundPolicyTitle = createInterpolateElement(
...
 	}
 );
 
 const paymentTitle = createInterpolateElement(
-	__( '<a>Safe & Secure</a> online payment', 'woocommerce' ),
+	__( '<a>Products</a> you can trust', 'woocommerce' ),
 	{
 		// eslint-disable-next-line jsx-a11y/anchor-has-content
-		a: <a href={ MARKETPLACE_URL + '/woocommerce-payments/' } />,
+		a: <a href={ MARKETPLACE_URL + '/products/' } />,
```
Comment (Dan-Q) [inline plugins/woocommerce-admin/client/marketplace/components/footer/footer.tsx:37]: ☝️ I've assumed this is the intended URL for this link.

Comment (kdevnel) [reply plugins/woocommerce-admin/client/marketplace/components/footer/footer.tsx:37]: Yeah, I'd say this is the correct URL.

**Review by kdevnel — APPROVED** [review]
Looks good! Thank you, Dan.

**Review by raicem — APPROVED** [review]
The marketplace page is getting better every day. Thank you @Dan-Q.

**Pull request merged**